src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/HttpClient/SendAsync.cs

```diff
@@ -127,23 +162,29 @@ private static void TryAttachHeadersToRequest(IAgent agent, HttpRequestMessage h
         }
 
         private static void TryProcessResponse(IAgent agent, Task<HttpResponseMessage> response, ITransaction transaction, ISegment segment, IExternalSegmentData externalSegmentData)
+        {
+
+            if (!ValidTaskResponse(response) || (segment == null))
+            {
+                return;
+            }
+
+            TryProcessResponse(agent, response?.Result, transaction, segment, externalSegmentData);
+        }
+
+        private static void TryProcessResponse(IAgent agent, HttpResponseMessage response, ITransaction transaction, ISegment segment, IExternalSegmentData externalSegmentData)
         {
             try
             {
-                if (!ValidTaskResponse(response) || (segment == null))
+                if (response == null || segment == null)
                 {
                     return;
                 }
 
-                var result = response?.Result;
+                var httpStatusCode = response.StatusCode;
+                externalSegmentData.SetHttpStatusCode((int)httpStatusCode);
```

Comment:
Why do you need the cast to `int` here?

Reply:
This is existed code.

Reply:
Ok, but why is the cast needed? The previous code had a `HasValue` check which you removed. We need to understand if this cast is still needed and why.

Reply:
The cast seems to be still needed (sooner or later) because the statuscode  attribute is type `long`. https://github.com/newrelic/newrelic-dotnet-agent/blob/2b8e8cdc71146b1196e6a455e981be37fe789532/src/Agent/NewRelic/Agent/Core/Segments/ExternalSegmentData.cs#L95

About the `HasValue` check removal. My refactoring makes sure the `response` object is never `null` when retrieving status code from this object.